### PR TITLE
CI build improvements

### PR DESCRIPTION
**Pull request by jaredpar**

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet-bot test this please

**Comment by jaredpar** [comment]
CC for review @agocke, @ellismg,

File: cibuild.sh [hunk 1]
```diff
@@ -61,7 +62,7 @@ run_msbuild()
     
     for i in `seq 1 $RETRY_COUNT`
     do
-        o=$(mono packages/Microsoft.Build.Mono.Debug.14.1.0.0-prerelease/lib/MSBuild.exe /v:m /p:SignAssembly=false /p:DebugSymbols=false "$@")
+        o=$(mono $MONO_ARGS packages/Microsoft.Build.Mono.Debug.14.1.0.0-prerelease/lib/MSBuild.exe /v:m /p:SignAssembly=false /p:DebugSymbols=false "$@")
```
Comment (agocke) [inline cibuild.sh]: I think we should discard this output capture. If the build fails I find it very useful to be able to look at the output and see why. Also, when the output is captured the build just sits there "Compiling CrossPlatform.sln" and I have no idea how far through it's gotten.

**Comment by agocke** [comment]
This looks good and contains a lot of the same fixes I made. I'm going to try to merge your PR into my PR and see how we do.

File: cibuild.sh (hunk 1 above)
Comment (jaredpar) [inline cibuild.sh]: That's fine.  I had hidden it when I was testing because I was losing the signal in the noise.  Can you undo this in your PR?

Comment (agocke) [inline cibuild.sh]: Already did.

**Comment by ellismg** [comment]
You might consider moving to MONO_ENV_OPTIONS environment variable so you don't have to remember to jam the options on every invocation.

**Pull request merged**